### PR TITLE
[Messenger] Deprecate option prefetch_count

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yesno
| Tickets       | -
| License       | MIT
| Doc PR        | -

Revert #30671

### CODE REVIEW

File: UPGRADE-5.3.md [hunk 1]
```diff
@@ -23,6 +23,11 @@ HttpKernel
 
  * Marked the class `Symfony\Component\HttpKernel\EventListener\DebugHandlersListener` as internal
 
+Messenger
+---------
+
+* Deprecated `prefetch_count` parameter, it has no effect and will be removed in Symfony 6.0.
```
Comment (chalasr) [inline UPGRADE-5.3.md]: Can you mention the concerned transport?

File: src/Symfony/Component/Messenger/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ CHANGELOG
 -----
 
  * `InMemoryTransport` can perform message serialization through dsn `in-memory://?serialize=true`.
+ * Deprecated `prefetch_count` parameter, it has no effect and will be removed in Symfony 6.0.
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/CHANGELOG.md]: This one should go in the amqp bridge's changelog I think

File: UPGRADE-5.3.md (hunk 1 above)
Comment (jderusse) [reply UPGRADE-5.3.md]: :+1:

File: src/Symfony/Component/Messenger/CHANGELOG.md (hunk 2 above)
Comment (jderusse) [reply src/Symfony/Component/Messenger/CHANGELOG.md]: right

**Review by chalasr — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Looks like there is a problem with this PR ("This commit does not belong to any branch on this repository."). Can you rebase please?

File: UPGRADE-6.0.md [hunk 3]
```diff
@@ -112,6 +112,8 @@ Messenger
  * Use of invalid options in Redis and AMQP connections now throws an error.
  * The signature of method `RetryStrategyInterface::isRetryable()` has been updated to `RetryStrategyInterface::isRetryable(Envelope $message, \Throwable $throwable = null)`.
  * The signature of method `RetryStrategyInterface::getWaitingTime()` has been updated to `RetryStrategyInterface::getWaitingTime(Envelope $message, \Throwable $throwable = null)`.
+ * Removed `prefetch_count` parameter in AMQP bridge.
+
```
Comment (xabbuh) [inline UPGRADE-6.0.md:116]: can be removed

File: src/Symfony/Component/Messenger/Bridge/Amqp/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+5.3.0
+-----
+
+* Deprecated `prefetch_count` parameter, it has no effect and will be removed in Symfony 6.0.
```
Comment (xabbuh) [inline src/Symfony/Component/Messenger/Bridge/Amqp/CHANGELOG.md]: Deprecated the [...]

File: UPGRADE-5.3.md [hunk 5]
```diff
@@ -23,6 +23,11 @@ HttpKernel
 
  * Marked the class `Symfony\Component\HttpKernel\EventListener\DebugHandlersListener` as internal
 
+Messenger
+---------
+
+* Deprecated `prefetch_count` parameter in AMQP bridge, it has no effect and will be removed in Symfony 6.0.
```
Comment (xabbuh) [inline UPGRADE-5.3.md]: Deprecated the [...] in the AMQP bridge [...]

File: UPGRADE-6.0.md [hunk 6]
```diff
@@ -112,6 +112,8 @@ Messenger
  * Use of invalid options in Redis and AMQP connections now throws an error.
  * The signature of method `RetryStrategyInterface::isRetryable()` has been updated to `RetryStrategyInterface::isRetryable(Envelope $message, \Throwable $throwable = null)`.
  * The signature of method `RetryStrategyInterface::getWaitingTime()` has been updated to `RetryStrategyInterface::getWaitingTime(Envelope $message, \Throwable $throwable = null)`.
+ * Removed `prefetch_count` parameter in AMQP bridge.
```
Comment (xabbuh) [inline UPGRADE-6.0.md]: Removed the `prefetch_count` parameter in the AMQP bridge.

**Review by xabbuh — APPROVED** [review]

File: UPGRADE-6.0.md (hunk 3 above)
Comment (jderusse) [reply UPGRADE-6.0.md:116]: Hmm. I'm not sure to understand why this removal should not be documented here?

Comment (jderusse) [reply UPGRADE-6.0.md:116]: Oh. the empty line!
Done

**Comment by chalasr** [comment]
Thank you @jderusse.

**Pull request merged**